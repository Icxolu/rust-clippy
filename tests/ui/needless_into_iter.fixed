//@run-rustfix

#![allow(unused)]
#![warn(clippy::needless_into_iter)]

fn main() {
    let sample = vec![0, 1, 2];
    let sample2 = vec!['t', 'e', 's', 't'];

    vec![].extend(Vec::<i32>::new());
    foo(Vec::new());
    bar(sample.clone().into_iter(), sample.clone());
    baz(sample, (), sample2);
}

fn foo(_: impl IntoIterator<Item = usize>) {}
fn bar<I: IntoIterator<Item = usize>>(_: impl Iterator<Item = usize>, _: I) {}
fn baz<I: IntoIterator<Item = usize>>(_: I, _: (), _: impl IntoIterator<Item = char>) {}
