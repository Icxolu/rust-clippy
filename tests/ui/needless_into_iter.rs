//@run-rustfix

#![allow(unused)]
#![warn(clippy::needless_into_iter)]

fn main() {
    let sample = vec![0, 1, 2];
    let sample2 = vec!['t', 'e', 's', 't'];

    vec![].extend(Vec::<i32>::new().into_iter());
    foo(Vec::new().into_iter());
    bar(sample.clone().into_iter(), sample.clone().into_iter());
    baz(sample, (), sample2.into_iter());
}

fn foo(_: impl IntoIterator<Item = usize>) {}
fn bar<I: IntoIterator<Item = usize>>(_: impl Iterator<Item = usize>, _: I) {}
fn baz<I: IntoIterator<Item = usize>>(_: I, _: (), _: impl IntoIterator<Item = char>) {}
