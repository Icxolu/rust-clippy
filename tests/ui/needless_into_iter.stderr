error: unneccessary `.into_iter()` call in generic argument
  --> $DIR/needless_into_iter.rs:10:36
   |
LL |     vec![].extend(Vec::<i32>::new().into_iter());
   |                                    ^^^^^^^^^^^^ help: remove this call
   |
   = note: `-D clippy::needless-into-iter` implied by `-D warnings`

error: unneccessary `.into_iter()` call in generic argument
  --> $DIR/needless_into_iter.rs:11:19
   |
LL |     foo(Vec::new().into_iter());
   |                   ^^^^^^^^^^^^ help: remove this call

error: unneccessary `.into_iter()` call in generic argument
  --> $DIR/needless_into_iter.rs:12:51
   |
LL |     bar(sample.clone().into_iter(), sample.clone().into_iter());
   |                                                   ^^^^^^^^^^^^ help: remove this call

error: unneccessary `.into_iter()` call in generic argument
  --> $DIR/needless_into_iter.rs:13:28
   |
LL |     baz(sample, (), sample2.into_iter());
   |                            ^^^^^^^^^^^^ help: remove this call

error: aborting due to 4 previous errors

